provision/kubernetes/deploy.go

```diff
@@ -433,17 +436,88 @@ func validateHC(hc *provTypes.TsuruYamlHealthcheck, client *ClusterClient) error
 		if hc.TimeoutSeconds >= hc.IntervalSeconds {
 			hc.IntervalSeconds = hc.TimeoutSeconds + 1
 		}
+
+		if err := ensureBackendConfig(ctx, client, app, hc); err != nil {
+			return err
+		}
+
 	}
 
 	return nil
 }
 
-func probesFromHC(hc *provTypes.TsuruYamlHealthcheck, client *ClusterClient, port int) (hcResult, error) {
+func int64PointerFromInt(v int) *int64 {
+	p := new(int64)
+	*p = int64(v)
+	return p
+}
+
+func ensureBackendConfig(ctx context.Context, client *ClusterClient, a provision.App, hc *provTypes.TsuruYamlHealthcheck) error {
```

Comment:
I think it would be nice for organization if we moved these functions related to backend config to a new file. Something like `gcp_backendconfig.go`.